### ISSUE TITLE
Proposal: `autocli` for osmosis

### ISSUE THREAD

**Issue by RusAkh** [osmosis-labs/osmosis#3227]
Suggestion: implement `autocli` features which are used in sdk 47 for osmosis.
* [Autocli usage in sdk](https://github.com/cosmos/cosmos-sdk/search?q=autocli)
* [Relevant issue](https://github.com/cosmos/cosmos-sdk/issues/11775)
* [Autocli godocs](https://pkg.go.dev/cosmossdk.io/api@v0.2.4/cosmos/autocli/v1)

**Comment by ValarDragon** [comment]
Yeah, very interested in doing this! I mentioned this to @faddat as something I'm really interested in!

Want to start trying this with `x/lockup` ?

**Comment by RusAkh** [comment]
Yes yes of course @ValarDragon

**Comment by RusAkh** [comment]
Autocli is a very undocumented feature so if anybody has some information about it or is willing to help, it would be fantastic!

**Comment by ValarDragon** [comment]
If its insufficient at any point, also open to forking it into osmosis & iterating here
